<launch>

	<!-- Command line arguments -->
	<arg name="map_file" />
	<arg name="num_robots" />
	<arg name="shared_dir" />

	<!-- Service/Topic/File Names -->
	<param name="box_plan_service"						type="string"	value="box_plan" />
	<param name="map_inflated_topic"					type="string"	value="/map_inflated" />
	<param name="box_size"								type="double"	value="0.4572" />
	<param name="grid_topic"							type="string"	value="/box/grid" />
	<param name="grid_marker_topic"						type="string"	value="/box/grid_marker" />
	<param name="grid_markerini_robot_markers_topic"	type="string"	value="/box/ini_robot_markers" />
	<param name="grid_markerini_boxes_markers_topic"	type="string"	value="/box/ini_boxes_markers" />
	<param name="grid_markercur_robot_markers_topic"	type="string"	value="/box/cur_robot_markers" />
	<param name="grid_markercur_boxes_markers_topic"	type="string"	value="/box/cur_boxes_markers" />
	<param name="grid_markerend_boxes_markers_topic"	type="string"	value="/box/end_boxes_markers" />
	<param name="shared_dir"							type="string"	value="$(arg shared_dir)" />
	<param name="goal_pos_file_format"					type="string"	value="%s/goal_pos_%03d" />
	<param name="current_pos_file_format"				type="string"	value="%s/current_pos_%03d" />

	<!-- Map server -->
	<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

	<!-- Map Inflator -->
	<node pkg="box" name="map_inflator"		type="map_inflator.py" respawn="false" output="screen">
		<param name="map_topic"				type="string"	value="/map" />
		<param name="inflation_ratio"		type="double"	value="0.5" />
	</node>

	<!-- Grid Mapper -->
	<node pkg="box" name="grid_mapper"		type="grid_mapper.py" respawn="false" output="screen">
		<param name="map_frame_id"			type="string"	value="/map" />
	</node>

	<!-- General Planner -->
	<node pkg="box" name="box_plan"	type="box_plan.py" respawn="false" output="screen">
		<param name="verbose"			type="bool"		value="false" />
		<param name="problem_csv"		type="string"	value="$(find box)/temp/problem_csv.csv" />
		<param name="planner_script"	type="string"	value="$(find box)/scripts/planning.sh" />
		<param name="solution"			type="string"	value="$(find box)/temp/solution" />
	</node>

	<!-- Execution Commander -->
	<node pkg="box" name="commander"	type="commander.py" respawn="false" output="screen">
		<param name="num_robots"			type="int"	value="$(arg num_robots)" />
	</node>

	<!-- Rviz -->
	<node name="rviz" pkg="rviz"		type="rviz" args="-d $(find box)/rviz/commander.rviz"/>

</launch>

